snorkel/parser.py

```diff
@@ -248,36 +284,198 @@ def parse_file(self, f, file_name):
+                tree = ' '.join(block['parse'].split())
+                if 'tree' not in document.meta:
+                    document.meta['tree'] = {}
+                document.meta['tree'][position] = tree
+
+            # Link the sentence to its parent document object
+            parts['document'] = document
+
+            # Add null entity array (matching null for CoreNLP)
+            parts['entity_cids'] = ['O' for _ in parts['words']]
+            parts['entity_types'] = ['O' for _ in parts['words']]
+
+            # Assign the stable id as document's stable id plus absolute character offset
+            abs_sent_offset_end = abs_sent_offset + parts['char_offsets'][-1] + len(parts['words'][-1])
+            parts['stable_id'] = construct_stable_id(document, 'sentence', abs_sent_offset, abs_sent_offset_end)
+            position += 1
+            yield parts
+
+
+class CoreNLPServer(object):
+    '''
+    See http://stanfordnlp.github.io/CoreNLP/corenlp-server.html
+
+    Spawn a StanfordCoreNLPServer process that accepts parsing requests at an HTTP port.
+    Kill it when python exits.
+    This makes sure that we load the models only once.
+    In addition, it appears that StanfordCoreNLPServer loads only required models on demand.
+    So it doesn't load e.g. coref models and the total (on-demand) initialization takes only 7 sec.
+    '''
+    def __init__(self, tokenize_whitespace=False, split_newline=False,
+                 parse_tree=False, strict_ptb=False, ptb3_escaping=True,
+                 annotators=['tokenize', 'ssplit', 'pos', 'lemma', 'depparse', 'ner'],
+                 annotator_opts={},
+                 java_xmx='4g', port=12345, num_threads=8, verbose=False, version='3.6.0'):
+        '''
+        Common configs:
+            1 sentence per line: ssplit.eolonly=True, tokenize.whitespace true
+
+        Advanced options can be passed in as a dictionary annotater_opts
+
+        :param tokenize_whitespace:
+        :param split_newline:
+        :param parse_tree:
+        :param strict_ptb:      (default:False) disable PTBTokenizer hacks
+        :param ptb3_escaping:    (default:True) enable PTB normalization
+        :param annotators:
+        :param annotater_opts:
+        :param java_xmx:
+        :param port:
+        '''
+        self.name = "CoreNLP"
+
+        # shortcut config flags
         self.parse_tree = parse_tree
+        self.strict_ptb = strict_ptb
+
+        # advanced config
+        self.tokenize_whitespace = tokenize_whitespace
+        self.split_newline = split_newline
+        self.annotators = annotators if not parse_tree else list(set(annotators + ['parse']))
+        self.annotater_opts = annotator_opts
+
+        self.port = port
+        self.timeout = 600000
+        self.num_threads = num_threads
+        self.verbose = verbose
+        self.version = version
+        self.block_elem = 'basicDependencies' if self.version == '3.7.0' else 'basic-dependencies'
```

Comment:
Yeah, this is a change just in 3.7.0 to the best of my knowledge. Only this element is changed from what I an tell. Refactored code to make the choice of parent name dependent on the specific version number.